when launching lots of tasks, the application failed to transfer message across hosts

Error log:
[ERROR] [04/08/2015 16:55:14.543] [OneForOneStrategy] Clients has not been launched properly before transporting messages
java.lang.Exception: Clients has not been launched properly before transporting messages
        at org.apache.gearpump.transport.Express.transport(Express.scala:107)
        at org.apache.gearpump.streaming.task.SendLater.flushPendingMessages(ExpressTransport.scala:105)
        at org.apache.gearpump.streaming.task.SendLater.org$apache$gearpump$streaming$task$SendLater$$sendPendingMessages(ExpressTransport.scala:88)
        at org.apache.gearpump.streaming.task.SendLater$$anonfun$sendAllPendingMsgs$1.apply(ExpressTransport.scala:110)
        at org.apache.gearpump.streaming.task.SendLater$$anonfun$sendAllPendingMsgs$1.apply(ExpressTransport.scala:110)
        at scala.collection.Iterator$class.foreach(Iterator.scala:750)
        at scala.collection.AbstractIterator.foreach(Iterator.scala:1202)
        at scala.collection.MapLike$DefaultKeySet.foreach(MapLike.scala:174)
        at org.apache.gearpump.streaming.task.SendLater.sendAllPendingMsgs(ExpressTransport.scala:110)
        at org.apache.gearpump.streaming.task.TaskActor$$anonfun$handleMessages$1.applyOrElse(TaskActor.scala:209)
        at akka.actor.Actor$class.aroundReceive(Actor.scala:465)
        at org.apache.gearpump.streaming.task.TaskActor.aroundReceive(TaskActor.scala:36)
        at akka.actor.ActorCell.receiveMessage(ActorCell.scala:516)
        at akka.actor.ActorCell.invoke(ActorCell.scala:487)
        at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:254)
        at akka.dispatch.Mailbox.run(Mailbox.scala:221)
        at akka.dispatch.Mailbox.exec(Mailbox.scala:231)
        at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
        at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
        at scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
        at scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)

--- a/core/src/main/scala/org/apache/gearpump/transport/Express.scala
+++ b/core/src/main/scala/org/apache/gearpump/transport/Express.scala
@@ -64,7 +64,7 @@ class Express(val system: ExtendedActorSystem) extends Extension with ActorLooku
   def startClients(hostPorts: Set[HostPort]): Future[Map[HostPort, ActorRef]] = {
     val clientsToClose = remoteClientMap.get().filterKeys(!hostPorts.contains(_)).keySet
     closeClients(clientsToClose)
-    hostPorts.filter(!localHost.equals(_)).map(hostPort =>
+    hostPorts.toList.filter(!localHost.equals(_)).foldLeft(Future(Map.empty[HostPort, ActorRef])) { (future, hostPort) =>
       remoteClientMap.alter { map =>
         if (!map.contains(hostPort)) {
           val actor = context.connect(hostPort)
@@ -72,7 +72,8 @@ class Express(val system: ExtendedActorSystem) extends Extension with ActorLooku
         } else {
           map
         }
-      }).lastOption.getOrElse(Future(Map.empty[HostPort, ActorRef]))
+      }
+    }
   }
 
   def closeClients(hostPorts: Set[HostPort]): Future[Map[HostPort, ActorRef]] = {

